src/server/test_utils.h

```diff
@@ -23,10 +23,18 @@ class TestConnection : public facade::Connection {
 
   void SendPubMessageAsync(PubMessage pmsg) final;
 
+  bool IsAdmin() const override {
+    return is_admin_;
+  }
+  void SetAdmin(bool is_admin) {
+    is_admin_ = is_admin;
+  }
+
```

Comment:
Do we still need these then? (also `is_admin_` member below)

Reply:
of course we need these. The IsAdmin is the Virtual function you suggested that we will use

Reply:
Right! sorry, I thought we're in a different class. My bad.